pkg/kubelet/cm/dra/manager.go

```diff
@@ -225,7 +225,7 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			for _, resourceHandle := range claimInfo.ResourceHandles {
 				// If no DriverName is provided in the resourceHandle, we
 				// use the DriverName from the status
-				pluginName := claimInfo.DriverName
+				pluginName := resourceHandle.DriverName
```

Comment:
You can remove the if check and update the comment. We require that resourceHandle.DriverName is set.

Reply:
https://github.com/kubernetes/kubernetes/pull/124075 clarified that in the documentation. The validation has always checked it.

Reply:
Thanks, will remove.

Reply:
It probably makes more sense to remove it in https://github.com/kubernetes/kubernetes/pull/124323 than here.

Reply:
Sure, I'll push all fixes into https://github.com/kubernetes/kubernetes/pull/124323 when I'm done with the e2e tests.

BTW, should we check if resourceHandle.DriverName is not empty or it's validated somewhere else?